File: 2006/J061.pas
const
  months: array[1..12] of string[3] = ('Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec');
var
  d, m, y, n, i: longint;
  mm: string[3];
  dummy: char;
begin
  read(d, dummy, mm, y);
  read(n);
  for i := 1 to 12 do
    if (months[i] = mm) then
      m := i;
  for i := 1 to n do
  begin
    inc(d);
    if ((d = 32) or (d = 31) and ((m = 4) or (m = 6) or (m = 9) or (m = 11))) then
    begin
      inc(m);
      d := 1;
    end;
    if ((d = 30) and (m = 2) or (d = 29) and (m = 2) and ((y mod 4 > 0) or (y mod 100 = 0) and (y mod 400 > 0))) then
    begin
      inc(m);
      d := 1
    end;
    if (m = 13) then
    begin
      inc(y);
      m := 1
    end;
  end;
  writeln(d, ' ', months[m], ' ', y)
end.

File: 2006/J061.cpp
#include <cstdio>
#include <cstring>
char months[12][4] = {"Jan", "Feb", "Mar", "Apr", "May",
    "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
int main() {
  int d, m, y, n;
  char mm[4];
  scanf("%d %s %d", &d, &mm, &y);
  scanf("%d", &n);
  for (int i = 0; i < 12; i++)
    if (strcmp(months[i], mm)==0)
      m = i + 1;
  for (int i = 0; i < n; i++) {
    d++;
    if (d == 32 || d == 31 && (m == 4 || m == 6 || m == 9 || m == 11)) {
      m++;
      d = 1;
    }
    if (d == 30 && m == 2 || d == 29 && m == 2 && (y % 4 != 0 || y % 100 == 0 && y%400 != 0)) {
      m++;
      d = 1;
    }
    if (m == 13) {
      y++;
      m = 1;
    }
  }
  printf("%d %s %d\n", d, months[m - 1], y);
  return 0;
}
